agent/engine/docker_task_engine_test.go

```diff
@@ -1515,3 +1517,75 @@ func TestCreateContainerOnAgentRestart(t *testing.T) {
 		t.Error("Unexpected error", metadata.Error)
 	}
 }
+
+func TestPullCNIImage(t *testing.T) {
+	ctrl, _, _, privateTaskEngine, _, _ := mocks(t, &config.Config{})
+	defer ctrl.Finish()
+	taskEngine, _ := privateTaskEngine.(*DockerTaskEngine)
+
+	container := &api.Container{
+		Type: api.ContainerCNIPause,
+	}
+	task := &api.Task{
+		Containers: []*api.Container{container},
+	}
+	metadata := taskEngine.pullContainer(task, container)
+	assert.Equal(t, DockerContainerMetadata{}, metadata, "expected empty metadata")
+}
+
+func TestPullNormalImage(t *testing.T) {
+	ctrl, client, _, privateTaskEngine, _, imageManager := mocks(t, &config.Config{})
+	defer ctrl.Finish()
+	taskEngine, _ := privateTaskEngine.(*DockerTaskEngine)
+	saver := mock_statemanager.NewMockStateManager(ctrl)
+	taskEngine.SetSaver(saver)
+
+	imageName := "image"
+	container := &api.Container{
+		Type:  api.ContainerNormal,
+		Image: imageName,
+	}
+	task := &api.Task{
+		Containers: []*api.Container{container},
+	}
+	imageState := &image.ImageState{
+		Image: &image.Image{ImageID: "id"},
+	}
+
+	client.EXPECT().PullImage(imageName, nil)
+	imageManager.EXPECT().RecordContainerReference(container)
+	imageManager.EXPECT().GetImageStateFromImageName(imageName).Return(imageState)
+	saver.EXPECT().Save()
+
+	metadata := taskEngine.pullContainer(task, container)
+	assert.Equal(t, DockerContainerMetadata{}, metadata, "expected empty metadata")
+}
+
+func TestPullEmptyVolumeImage(t *testing.T) {
+	ctrl, client, _, privateTaskEngine, _, _ := mocks(t, &config.Config{})
+	defer ctrl.Finish()
+	taskEngine, _ := privateTaskEngine.(*DockerTaskEngine)
+	saver := mock_statemanager.NewMockStateManager(ctrl)
+	taskEngine.SetSaver(saver)
+
+	imageName := "image"
+	container := &api.Container{
+		Type:  api.ContainerEmptyHostVolume,
+		Image: imageName,
+	}
+	task := &api.Task{
+		Containers: []*api.Container{container},
+	}
+
+	// Windows is special-cased here and just performs a normal pull
+	if runtime.GOOS == "windows" {
```

Comment:
I'd prefer this test to be in platform specific files if the expectations are different.

Reply:
Yeah, wasn't sure about that since this is the only difference in all these tests.  I can move it though.

Reply:
> Have this check at the top of the function and do a t.Skip() if you don't want to make a separate Linux file?

There's both a Linux- and Windows-test, sharing most of the test structure except some of the expectations.

> You could also make this more OO instead of using platform constants.

Not sure I understand what you mean here.

Reply:
You branch based on a constant boolean that's defined for each platform. You could instead attach these functions to a struct with this configuration embedded.

The tests would then be driven by configuration instead of driven by platform.

Reply:
I went and separated them into platform-specific files.